pkg/lib/config/authentication_flow_bot_protection.go

```diff
@@ -0,0 +1,60 @@
+package config
+
+var _ = Schema.Add("AuthenticationFlowBotProtection", `
+{
+	"type": "object",
+	"additionalProperties": false,
+	"required": ["mode"],
+	"properties": {
+		"mode": { "type": "string", "enum": ["never", "always"] }
+	},
+	"allOf": [
+		{
+			"if": {
+				"required": ["mode"],
+				"properties": {
+					"mode": {
+						"enum": ["always"]
+					}
+				}
+			},
+			"then": {
+				"required": ["provider"],
+				"properties": {
+					"provider": { "$ref": "#/$defs/AuthenticationFlowBotProtectionProvider" }
+				}
+			}
+		}
+	]
+}
+`)
+
+var _ = Schema.Add("AuthenticationFlowBotProtectionProvider", `
+{
+	"type": "object",
+	"additionalProperties": false,
+	"required": ["type"],
+	"properties": {
+		"type": { "type": "string", "enum": ["cloudflare", "recaptchav2"] }
+	}
+}
+`)
+
+type AuthenticationFlowBotProtectionProvider struct {
+	Type BotProtectionProviderType `json:"type,omitempty"`
+}
+type AuthenticationFlowBotProtection struct {
+	Mode     AuthenticationFlowBotProtectionMode      `json:"mode,omitempty"`
+	Provider *AuthenticationFlowBotProtectionProvider `json:"provider,omitempty" nullable:"true"`
```

Comment:
Do not add nullable here. Add it to where `AuthenticationFlowBotProtection` is used in a struct.